File: Algorithms/ Sorting/Insertion Sort - Part 1.php
<?php

/*
 * Complete the 'insertionSort1' function below.
 *
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. INTEGER_ARRAY arr
 */

function insertionSort1($n, $arr) {
    # Write your code here
    $key = $arr[$n - 1];
    $j = $n - 2;
    while ($j >= 0 && $arr[$j] > $key) {
        $arr[$j + 1] = $arr[$j];
        echo implode(' ', $arr) . "\n";
        $j--;
    }
    $arr[$j + 1] = $key;
    echo implode(' ', $arr) . "\n";
}

$n = intval(trim(fgets(STDIN)));

$arr_temp = rtrim(fgets(STDIN));

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

insertionSort1($n, $arr);

File: Algorithms/ Sorting/Insertion Sort - Part 1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'insertionSort1' function below.
     *
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. INTEGER_ARRAY arr
     */

    public static void insertionSort1(int n, List<int> arr)
    {
        int key = arr[n - 1];
        int j = n - 2;
        while (j >= 0 && arr[j] > key)
        {
            arr[j + 1] = arr[j];
            Console.WriteLine(string.Join(" ", arr));
            j--;
        }
        arr[j + 1] = key;
        Console.WriteLine(string.Join(" ", arr));
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> arr = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(arrTemp => Convert.ToInt32(arrTemp)).ToList();

        Result.insertionSort1(n, arr);
    }
}
